stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -2,30 +2,315 @@
 
 package com.facebook.stetho.inspector.protocol.module;
 
+import android.util.SparseArray;
+
+import com.facebook.stetho.common.Util;
+import com.facebook.stetho.inspector.helper.ChromePeerManager;
 import com.facebook.stetho.inspector.jsonrpc.JsonRpcPeer;
 import com.facebook.stetho.inspector.jsonrpc.JsonRpcResult;
 import com.facebook.stetho.inspector.protocol.ChromeDevtoolsDomain;
 import com.facebook.stetho.inspector.protocol.ChromeDevtoolsMethod;
+import com.facebook.stetho.json.ObjectMapper;
 import com.facebook.stetho.json.annotation.JsonProperty;
 
 import org.json.JSONObject;
 
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+
 public class DOM implements ChromeDevtoolsDomain {
-  public DOM() {
+  private final Provider mDOMProvider;
+  private final DOMPeerManager mPeerManager;
+  private final ObjectMapper mObjectMapper;
+
+  private final Object mNodeMapSync = new Object();
```

Comment:
This can be better expressed by a separate class which handles `mNextNodeId`, `mElementToNodeIdMap`, etc.  Easier to reason about and can be re-used later by `Console`.  Something like `ObjectIdMapper`.  Could be put into `inspector/helper`.